app/controllers/meal_attendances_controller.rb

```diff
@@ -2,8 +2,21 @@
 
 class MealAttendancesController < ApplicationController
   before_action :find_attendance_by_id, only: %i[edit update show destroy]
+  before_action :authenticate_user!
+  load_and_authorize_resource
   def index
-    @MealAttendances = MealAttendance.all
+    @today_lunch_count=MealAttendance.where("DATE(meal_date)=? AND meal_type=?", Date.today,0).count
+    @today_snack_count=MealAttendance.where("DATE(meal_date)=? AND meal_type=?", Date.today,1).count
+
+    new_query={}
+    new_query.merge!(meal_type: params[:category]) if params[:category].present? && params[:category]!=''
+    new_query.merge!(meal_date: params[:search_date]) if params[:search_date].present? && params[:search_date]!=''
+
+    new_query.compact_blank
+    new_query.compact!
+
+    @MealAttendances=MealAttendance.where(new_query)
+    @MealAttendances=@MealAttendances.joins(:user).where("first_name LIKE ?", "%#{params[:search].delete(' ')}%") if params[:search].present? && params[:search]!=''
```

Comment:
Use strip for avoid leading and trailing spaces.
```suggestion
    if params[:search].present? && params[:search].strip != ''
       search_query = params[:search].delete(' ')
       @MealAttendances = @MealAttendances.joins(:user).where("first_name LIKE ?", "%#{search_query}%")
    end
```